gcloud-java-contrib/gcloud-java-nio/README.md

```diff
@@ -0,0 +1,210 @@
+and guarantees consistency: when a write succeeds, the latest copy of the
+object will be returned to any GET, globally.
+
+See the [Google Cloud Storage docs][cloud-storage-activation] for more details
+on how to activate Cloud Storage for your project.
+
+About Java NIO Providers
+------------------------
+
+Java NIO Providers is an extension mechanism that is part of Java and allows
+third parties to extend Java's [normal File API][java-file-api] to support
+additional filesystems.
+
+Getting Started
+---------------
+#### Prerequisites
+
+For this tutorial, you will need a [Google Developers
+Console](https://console.developers.google.com/) project with "Google Cloud
+Storage" and "Google Cloud Storage JSON API" enabled via the console's API
+Manager. You will need to [enable
+billing](https://support.google.com/cloud/answer/6158867?hl=en) to use Google
+Cloud Storage. [Follow these
+instructions](https://cloud.google.com/docs/authentication#preparation) to get
+your project set up. You will also need to set up the local development
+environment by [installing the Google Cloud SDK](https://cloud.google.com/sdk/)
+and running the following commands in command line: `gcloud auth login` and
+`gcloud config set project [YOUR PROJECT ID]`.
+
+#### Installation and setup
+You'll need to obtain the `gcloud-java-nio` library.
+
+There are two ways to use this library.
+
+The recommended way is to follow the [Quickstart](#quickstart) section to add
+`gcloud-java-nio` as a dependency in your code.
+
+The second way is more complicated, but it allows you to add Google Cloud
+Storage support to some legacy Java programs. This approach is described in the
+[gcloud-java-nio-examples README](../gcloud-java-nio-examples/README.md).
+
+#### Accessing files
+
+The simplest way to get started is with `Paths` and `Files`:
+
+    Path path = Paths.get(URI.create("gs://bucket/lolcat.csv"));
+    List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+
+If you know the paths will point to Google Cloud Storage, you can also use the
+direct formulation:
+
+    try (CloudStorageFileSystem fs = CloudStorageFileSystem.forBucket("bucket") {
+      Path path = fs.getPath("lolcat.csv");
+      List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+    }
+
+Once you have a `Path` you can use it as you would a normal file. For example
+you can use `InputStream` and `OutputStream` for streaming:
+
+    try (InputStream input = Files.openInputStream(path)) {
+      // ...
+    }
+
+You can also set various attributes using CloudStorageOptions static helpers:
+
+    Files.write(csvPath, csvLines, StandardCharsets.UTF_8,
+        withMimeType(MediaType.CSV_UTF8),
+        withoutCaching());
+
+Limitations
+-----------
+
+This library is usable, but not yet complete. The following features are not
+yet implemented:
+ * Listing all the buckets
+ * Resuming upload or download
+ * Generations
+ * File attributes
+ * (more)
```

Comment:
Before we merge perhaps we should put something more explicit, like "(list is not exhaustive)"?
(yes I know I'm the one who put "more" in there in the first place!)